source/base/timer.cc

```diff
@@ -353,8 +353,13 @@ TimerOutput::TimerOutput (MPI_Comm      mpi_communicator,
 
 TimerOutput::~TimerOutput()
 {
-  while (active_sections.size() > 0)
-    leave_subsection();
+  try
+    {
+      while (active_sections.size() > 0)
+        leave_subsection();
```

Comment:
Then we would not be able to catch any exceptions that triggered the destruction of this object. What about abort if we catch `std::exception &` and otherwise ignore a thrown exception?